store/tikv/coprocessor.go

```diff
@@ -664,7 +708,14 @@ func (it *copIterator) Next(ctx context.Context) (kv.ResultSubset, error) {
 			// Switch to next task.
 			it.tasks[it.curr] = nil
 			it.curr++
-			it.sendRate.putToken()
+			it.actionOnExceed.cond.L.Lock()
+			// The tasks whose id is less than maxID are assumed that being sending to their task channel.
+			// So the response channel would be thought as drained out if the current taskID is greater or equal than
+			// the maxID as all the workers are being suspended at that time.
+			if it.actionOnExceed.cond.exceed && it.tasks[it.curr].id >= it.actionOnExceed.cond.maxID {
+				broadcast()
+			}
+			it.actionOnExceed.cond.L.Unlock()
```

Comment:
Do we need to put token back here?

Reply:
The `putToken` logic now handled by the copWorker in its `run()` function. After copWorker finished one task each time, the send rate would put token without considering `keepOrder` or not, unless the oom action happened.